openpype/hosts/traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -0,0 +1,682 @@
+                        relative_thumbnail_path = repre_data["thumbnailPath"]
+                        # representation might not have thumbnail path
+                        # so ignore this one
+                        if not relative_thumbnail_path:
+                            continue
+                        thumb_dir, thumb_file = \
+                            self._get_refactor_thumbnail_path(
+                                staging_dir, relative_thumbnail_path)
+                        filename, ext = os.path.splitext(thumb_file)
+                        thumbnail_repr_data = deepcopy(
+                            base_thumbnail_repre_data)
+                        thumbnail_repr_data.update({
+                            "name": "thumbnail_{}".format(filename),
+                            "ext": ext[1:],
+                            "files": thumb_file,
+                            "stagingDir": thumb_dir,
+                            "outputName": explicit_output_name,
+                        })
+                        new_instance["prepared_data_for_repres"].append(
+                            ("_thumbnail_", thumbnail_repr_data)
+                        )
+                    elif (
+                        thumbnails
+                        and not multiple_thumbnails
+                        and not thumbnails_processed
+                        or not reviewable
+                    ):
+                        # here we will use only one thumbnail for
+                        # all representations
+                        relative_thumbnail_path = repre_data["thumbnailPath"]
+                        if not relative_thumbnail_path:
+                            relative_thumbnail_path = thumbnails.pop()
+                        thumb_dir, thumb_file = \
+                            self._get_refactor_thumbnail_path(
+                                staging_dir, relative_thumbnail_path)
+                        _, ext = os.path.splitext(thumb_file)
+                        thumbnail_repr_data = deepcopy(
+                            base_thumbnail_repre_data)
+                        thumbnail_repr_data.update({
+                            "ext": ext[1:],
+                            "files": thumb_file,
+                            "stagingDir": thumb_dir
+                        })
+                        new_instance["prepared_data_for_repres"].append(
+                            ("_thumbnail_", thumbnail_repr_data)
+                        )
+                        thumbnails_processed = True
+
+                    # get representation data
+                    representation_data = self._get_representation_data(
+                        filepath, repre_data, staging_dir,
+                        explicit_output_name
+                    )
+
+                    new_instance["prepared_data_for_repres"].append(
+                        (repre_data["color"], representation_data)
```

Comment:
Maybe it should be colorspace?

`(repre_data["colorspace"], representation_data)`

Reply:
Yes, quite right. Fixed here https://github.com/ynput/ayon-core/pull/100/files#diff-05b3c0cf629518bc7e7233fb6765bd2c9412150a2c9582edf75e086954673a3bR284-R286